pkg/beyla/network_cfg.go

```diff
@@ -33,6 +33,9 @@ type NetworkConfig struct {
 	// Default value is false (disabled)
 	Enable bool `yaml:"enable" env:"BEYLA_NETWORK_METRICS"`
 
+	// Don't cleanup QDisc on start/stop. This helps ensure Beyla can work with Cilium CNI.
+	WithoutTC bool `yaml:"no_tc" env:"BEYLA_NETWORK_NO_TC"`
```

Comment:
I'd maybe replace this boolean by a string enum. Something like: `Source` whose values would be `tc` or `socket`.

Reply:
Oh, that's an amazing suggestion! I'm going to work on adding this!